rllib/offline/offline_data.py

```diff
@@ -206,14 +202,12 @@ def sample(
         # returned now and we have already generated from the iterator, i.e.
         # `isinstance(self.batch_iterators, types.GeneratorType) == True`, we need
         # to create here a new iterator.
-        # TODO (simon): Check, if this iterator could potentially exhaust.
```

Comment:
Does that 100% not exhaust?

Reply:
You left another comment further down 
"# If the batch iterator is exhausted, reinitiate a new one.".
So to me it looks like you knew that it would be exhausted? It's part of the design here, no?